### PR TITLE
rbac: Refactor RBAC dependencies

### PR DESCRIPTION
**Pull request by jkosh44**
This commit refactors the mz_adapter::rbac dependencies so that it no
longer relies on mz_adapter modules. This will allow us to move
mz_adapter::rbac into the mz_sql crate, which will help remove
mz_adapter dependencies from mz_adapter::catalog::storage which will
help in the effort to extract catalog::storage into its own crate.

Works towards resolving #20953

### Motivation
This PR refactors existing code.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

**Comment by jkosh44** [comment]
An alternative approach is here: https://github.com/MaterializeInc/materialize/pull/21631

File: src/sql/src/rbac.rs [hunk 1]
```diff
@@ -1499,15 +1512,18 @@ pub(crate) const fn all_object_privileges(object_type: SystemObjectType) -> AclM
...
         grantee: owner_id,
         grantor: owner_id,
         acl_mode: all_object_privileges(SystemObjectType::Object(object_type)),
     }
 }
 
-pub(crate) const fn default_builtin_object_privilege(object_type: ObjectType) -> MzAclItem {
+pub const fn default_builtin_object_privilege(
+    object_type: ObjectType,
+    mz_system_role_id: RoleId,
```
Comment (ParkMyCar) [inline src/sql/src/rbac.rs:1525]: I'm on the fence about this, but I feel like `MZ_SYSTEM_ROLE_ID` could also get moved to `mz_sql`. Not a huge deal, but would prevent needing to take it as an arg here

**Review by ParkMyCar — APPROVED** [review]
Nice! I'm glad this worked out!

**Comment by ParkMyCar** [comment]
> An alternative approach is here: #21631

Thank for trying the alternative! Personally I like this approach much more

File: src/sql/src/rbac.rs (hunk 1 above)
Comment (jkosh44) [reply src/sql/src/rbac.rs]: I moved it, I agree that it's a bit cleaner.

**Review by ParkMyCar — APPROVED** [review]
Nice, thanks for moving the IDs!

**Pull request merged**